src/libxively/control_topic/xi_control_topic_layer.c

```diff
@@ -78,22 +76,49 @@ xi_control_topic_subscribe( void* context, char* subscribe_control_topic_name );
 static xi_state_t
 xi_control_topic_connection_state_changed( void* context, xi_state_t state );
 
-#if ( 0 )
-/*
- * This is unused at this time.
- */
-static xi_state_t xi_control_topic_publish_on_topic( void* context, xi_data_desc_t* data )
+static xi_state_t
+xi_control_topic_publish_on_topic( void* context, xi_control_message_t* control_message )
 {
+    // printf( "%s, control_message: %p\n", __FUNCTION__, control_message );
+
     xi_state_t local_state = XI_STATE_OK;
 
     xi_control_topic_layer_data_t* layer_data =
         ( xi_control_topic_layer_data_t* )XI_THIS_LAYER( context )->user_data;
 
-    assert( layer_data != NULL );
+    assert( NULL != layer_data /* && NULL != data */ );
+    if ( NULL == control_message )
+    {
+        return XI_INVALID_PARAMETER;
+    }
```

Comment:
error handling. This was a topic we disputed about with Olgierd. Olgierd likes assertion. I like it less. Here I wouldn't crash the program just because somebody wants to send a NULL control message. I think this situation can be handled: it is a no-operation.
What I did is I moved this condition to be the first action in the function. Hope this excuse is ok for you :) (No is a good answer here :) )